### ISSUE TITLE
Intellisense doesn't appear to work in Debug Interactive console

### ISSUE THREAD

**Issue by bsdz** [microsoft/PTVS#2953]
Variable completion (IntelliSense) does not appear to work in the Debug Interactive window. If I try to tab to see a list of members, the console just inserts a tab character. I've tried various combinations of control+space, control+j etc. 

Interestingly, if I switch the console's Environment to either a "Project: .." or the name of an existing Python environment, then completion behaves as expected. I've searched through documentation, open/closed issues to determine if this is a known bug or feature but nothing came up. 

[Here's a gif demonstrating the issue](https://thumbs.gfycat.com/YellowBriefAppaloosa-size_restricted.gif). 

Perhaps I'm missing something very simple? Below is my install details.

<details>
<summary>Visual Studio Install Info</summary>
Microsoft Visual Studio Community 2017 
Version 15.3.2
VisualStudio.15.Release/15.3.2+26730.10
Microsoft .NET Framework
Version 4.7.02531

Installed Version: Community

Visual Basic 2017   00369-60000-00001-AA642
Microsoft Visual Basic 2017

Visual C# 2017   00369-60000-00001-AA642
Microsoft Visual C# 2017

Visual C++ 2017   00369-60000-00001-AA642
Microsoft Visual C++ 2017

Application Insights Tools for Visual Studio Package   8.8.00712.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2017   15.0.30726.0
ASP.NET and Web Tools 2017

ASP.NET Core Razor Language Services   1.0
Provides languages services for ASP.NET Core Razor.

ASP.NET Template Engine 2017   15.0.30726.0
ASP.NET Template Engine 2017

Azure App Service Tools v3.0.0   15.0.30728.0
Azure App Service Tools v3.0.0

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Cookiecutter   3.0.17208.1
Provides tools for finding, instantiating and customizing templates in cookiecutter format.

GitHub.VisualStudio   2.3.0.24
A Visual Studio Extension that brings the GitHub Flow into Visual Studio.

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

JavaScript UWP Project System   2.0
JavaScript UWP Project System

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2017 - v2.9.50719.1

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   4.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

Python   3.0.17208.1
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.

Python - Django support   3.0.17208.1
Provides templates and integration for the Django web framework.

Python - IronPython support   3.0.17208.1
Provides templates and integration for IronPython-based projects.

Python - Profiling support   3.0.17208.1
Profiling support for Python projects.

TypeScript   2.3.4.0
TypeScript tools for Visual Studio

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio tools for CMake   1.0
Visual Studio tools for CMake

Visual Studio Tools for Universal Windows Apps   15.0.26730.08
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.
</details>

**Comment by huguesv** [comment]
You're not missing anything, it's a known issue.

The completions based on analysis have problems: https://github.com/Microsoft/PTVS/issues/1788

The live completions won't work because the necessary messages for the new protocol haven't been implemented (something was preventing them from even being called with the old protocol).

**Comment by zooba** [comment]
Also reported at https://developercommunity.visualstudio.com/content/problem/58731/intellisense-does-not-work-in-after-breakpoint-in.html

**Comment by zooba** [comment]
This is actually just [not implemented](https://github.com/Microsoft/PTVS/blob/b756e79ca9dd6a01a835bed3dc57fb32d800b0d4/Python/Product/PythonTools/PythonTools/Repl/PythonDebugProcessReplEvaluator.cs#L253-L256) at all - we have some work to do to wrap up our AD7 engine to collect the names, I guess.

**Comment by huguesv** [comment]
The not implemented code was never getting called, due to other bugs / asserts, which is why I postponed it. If you're working on #2065 and it unblocks this one, I can take over from there.

**Comment by zooba** [comment]
I've also got half of this one done (at least the easy/annoying case of locals), since I wanted to get the current filename so we get the right analyzer as part of the other fix. As for getting members and signatures, you're welcome to take over that part.

**Comment by huguesv** [comment]
I made some minor improvements so that `obj.` or `module.` will now get completions when evaluating against the current frame.
https://github.com/Microsoft/PTVS/pull/3342

We'll hold off on the remaining work until we've made some more progress on the planned debugger improvements and have a clearer picture of what we have available for implementing these IntelliSense features.

**Comment by zooba** [comment]
Also reported at https://developercommunity.visualstudio.com/content/problem/171611/python-intellisense-dont-work-in-debug-window.html

**Comment by nambread** [comment]
I'm also experiencing this issue in VS 2019 Professional. I've connected a remote debugger to a program (Autodesk Maya) and the code completion doesn't seem to work. (Ctrl+Space on `obj.` or `module.`)

@huguesv 's comment above seems to imply a partial update was merged down for this. Has there been a regression?
